### PR TITLE
migrate kube-lego to cert-manager

### PR DESCRIPTION
**Pull request by krzyzacy**
ref https://github.com/kubernetes/test-infra/issues/9706
kube-lego has been deprecated since 1.8

I took https://github.com/jetstack/cert-manager/blob/master/contrib/manifests/cert-manager/with-rbac.yaml and added [a cluster issuer](https://cert-manager.readthedocs.io/en/latest/reference/clusterissuers.html) and [the ingress shim](https://cert-manager.readthedocs.io/en/latest/reference/ingress-shim.html)

(our cert expires in next Jan so we should be fine...)

/assign @munnerz @cjwagner @fejta @stevekuznetsov

### CODE REVIEW

File: prow/cluster/cert-manager-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,259 @@
+---
```
Comment (fejta) [inline prow/cluster/cert-manager-deployment.yaml]: Looks like this is more than just the deployment. So can you rename this to just cert-manager.yaml?

File: prow/cluster/cert-manager-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+# Source: cert-manager/templates/certificate-crd.yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: certificates.certmanager.k8s.io
+  annotations:
+    "helm.sh/hook": crd-install
+  labels:
+    app: cert-manager
+    chart: cert-manager-v0.6.0-dev.2
```
Comment (munnerz) [inline prow/cluster/cert-manager-deployment.yaml]: Looks like you've pulled these manifests from the master branch - you should be using `release-0.5` instead, unless you're feeling super :this_is_fine: 😄

File: prow/cluster/cert-manager-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  acme:
+    # The ACME server URL
+    server: https://acme-v01.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: spxtr@google.com
+    # Name of a secret used to store the ACME account private key from step 3
+    privateKeySecretRef:
+      name: letsencrypt-private-key
+    # Enable the HTTP-01 challenge provider
+    http01: {}
```
Comment (munnerz) [inline prow/cluster/cert-manager-deployment.yaml]: nit: The ClusterIssuer might be better off in a separate file to make upgrades easier in future

File: prow/cluster/tls-ing_ingress.yaml [hunk 4]
```diff
@@ -19,6 +19,8 @@ metadata:
   annotations:
     kubernetes.io/ingress.global-static-ip-name: prow
     kubernetes.io/tls-acme: "true"
+    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
+    certmanager.k8s.io/acme-http01-edit-in-place: "true"
```
Comment (munnerz) [inline prow/cluster/tls-ing_ingress.yaml]: 👍 assuming you are using GCE LB, this annotation is correct

File: prow/cluster/cert-manager.yaml [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  acme:
+    # The ACME server URL
+    server: https://acme-v01.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: spxtr@google.com
+    # Name of a secret used to store the ACME account private key from step 3
+    privateKeySecretRef:
+      name: letsencrypt-private-key
+    # Enable the HTTP-01 challenge provider
+    http01: {}
```
Comment (munnerz) [inline prow/cluster/cert-manager.yaml]: ClusterIssuer should probably be in a different file (and if you want to keep it here, you are missing the `---` separator)

File: prow/cluster/cert-manager.yaml [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+---
+# Source: cert-manager/templates/deployment.yaml
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: cert-manager
+  namespace: "cert-manager"
+  labels:
+    app: cert-manager
+    chart: cert-manager-v0.6.0-dev.2
```
Comment (munnerz) [inline prow/cluster/cert-manager.yaml]: Looks like you've pulled the manifests from master instead of `release-0.5` 😄

**Comment by munnerz** [comment]
sorry about the duplicate comments - perhaps due to new GitHub UI changes I got a weird error half way through review 😓

File: prow/cluster/cert-manager-deployment.yaml (hunk 1 above)
Comment (fejta) [reply prow/cluster/cert-manager-deployment.yaml]: (Follow the getting_started and grandmatriarch patterns)

Comment (krzyzacy) [reply prow/cluster/cert-manager-deployment.yaml]: yeah got it, there's `MULTI_KIND` :-)

File: prow/cluster/cert-manager.yaml (hunk 6 above)
Comment (krzyzacy) [reply prow/cluster/cert-manager.yaml]: yep, switched back to 0.5.0 :-)

File: prow/cluster/cert-manager.yaml (hunk 5 above)
Comment (krzyzacy) [reply prow/cluster/cert-manager.yaml]: put issuer to a separate file

**Comment by krzyzacy** [comment]
(also updated docs)

File: prow/cluster/cluster-issuer.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+# cluster-issuer
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: ClusterIssuer
+metadata:
+  # Adjust the name here accordingly
+  name: letsencrypt-prod
+spec:
+  acme:
+    # The ACME server URL
+    server: https://acme-v01.api.letsencrypt.org/directory
```
Comment (munnerz) [inline prow/cluster/cluster-issuer.yaml]: This should be acme-v02 now. cert-manager v0.5.0 does not support v01 endpoints
```suggestion
    server: https://acme-v02.api.letsencrypt.org/directory
```

Comment (krzyzacy) [reply prow/cluster/cluster-issuer.yaml]: ah thanks, this one was copied from kube-lego :see_no_evil:

**Comment by krzyzacy** [comment]
note here to next prow bumper: the bazel release workflow won't clean up kube-lego deployment, that need to be done manually (the kube-lego deployment, and the account secret in default ns)

**Comment by fejta** [comment]
Maybe remove it from the bazel target but keep the file (maybe as a separate target)? Then we can delete it easily. Or we could just check out an older commit.

Also let me know if you want me to just lgtm... I don't really know the details of cert-manager and would prefer munnerz review that part but happy to move things along if necessary?

**Comment by krzyzacy** [comment]
I don't think have a file or not will make a difference since kube-lego creates magics in the cluster, I'll be able to handle that.

@munnerz any more comments? :-)

**Comment by krzyzacy** [comment]
/hold cancel
/this-is-fine
ok I'll have some fun today (cc @amwat)

**Pull request merged**